test/widgets/message_list_test.dart

```diff
@@ -923,6 +923,53 @@ void main() {
         await tester.pump();
         tester.widget(find.text('new stream name'));
       });
+
+      testWidgets('navigates to TopicNarrow on tapping topic in ChannelNarrow', (tester) async {
+        final pushedRoutes = <Route<void>>[];
+        final navObserver = TestNavigatorObserver()
+          ..onPushed = (route, prevRoute) => pushedRoutes.add(route);
+        final channel = eg.stream();
+        await setupMessageListPage(tester,
+          narrow: ChannelNarrow(channel.streamId),
+          messages: [eg.streamMessage(stream: channel, topic: 'topic name')],
+          streams: [channel],
+          navObservers: [navObserver]);
```

Comment:
Cool, yeah, putting the `tester` argument on the same line as the function and the close-paren on the same line as the last argument are examples of what I had in mind at https://github.com/zulip/zulip-flutter/pull/1193#discussion_r1926387133 .